experiment/resultstore/convert.go

```diff
@@ -114,6 +115,13 @@ func convert(project, details string, url gcs.Path, result downloadResult) (resu
 	artifactsPath := basePath + "artifacts/"
 	buildLog := basePath + "build-log.txt"
 	bucket := url.Bucket()
+
+	pathSegments := strings.Split(strings.TrimSuffix(url.String(), "/"), "/")
```

Comment:
Can we centralize the parsing of an opaque string into a job name? Is this the reverse of the logic in the pod utils library and is there already a transform there? Can we reuse what the spyglass code does?

Reply:
Thanks Steve. With Katharine's help I found the parsing logic in Spyglass. However, it's a method of the Spyglass struct that requires instantiation of a bunch of things that this experiment dir doesn't use. I refactored this into a standalone function, which is very similar to the Spyglass logic now (simplified).

Katharine is working on https://github.com/kubernetes/test-infra/pull/15732, once that's merged, I'll update this to directly fetch the name from the JSON artifact instead of parsing the URL string. For now, let's have this separate logic in this experimental dir. What do you think?

Reply:
If the standalone functions are similar, let's put them in a library and call them from there instead of having two similar but different implementations.

Reply:
Given that this function is trivial, the Spyglass version does substantially more work, this is an experiment, and the necessity of having this function here at all should be short-lived (due to #15785), I don't think this PR should require factoring this out of Spyglass.

Reply:
Thanks Katharine. +1. Atm, the 2 implementations are very different.
Thanks Steve for suggesting. We can revisit this when we formalize the use of resultstore, when Katharine's change in merged, there will be a better way to tackle this.